src/client/mod.rs

```diff
@@ -141,27 +146,32 @@ impl APIClient {
         }
     }
 
-    pub async fn request_events<T>(&self, request: http::Request<Vec<u8>>) -> Result<Vec<T>>
+    pub async fn request_events<T>(
+        &self,
+        request: http::Request<Vec<u8>>,
+    ) -> Result<impl Stream<Item = Result<T>>>
     where
         T: DeserializeOwned,
     {
-        let res : reqwest::Response = self.send(request).await?;
+        let res: reqwest::Response = self.send(request).await?;
         trace!("{} {}", res.status().as_str(), res.url());
-        //trace!("Response Headers: {:?}", res.headers());
-        let s = res.status();
-        let text = res.text().await?;
-        handle_api_errors(&text, &s)?;
 
-        // Should be able to coerce result into Vec<T> at this point
-        let mut xs : Vec<T> = vec![];
-        for l in text.lines() {
-            let r = serde_json::from_str(&l).map_err(|e| {
-                warn!("{} {:?}", l, e);
-                Error::SerdeError(e)
-            })?;
-            xs.push(r);
-        }
-        Ok(xs)
+        // Now use `unfold` to convert the chunked responses into a Stream
+        Ok(futures::stream::unfold(res, |mut resp| {
+            async move {
```

Comment:
looks like you only wrapped this in an Ok and did an async move here?

Reply:
The `Ok` is for the so we can potentially terminate earlier above with:
 ```rust
let res: reqwest::Response = self.send(request).await?;
```
I was going to change the signature to just `impl Stream<Item = Result<T>>` but wasn't sure what to return if the first request fails, an empty stream felt not quite right.

Reply:
Ah, that makes sense. I wasn't sure how to shoehorn in the first Result in the signature.